src/Microsoft.Management.Deployment/PackageCatalogProgress.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "PackageCatalogProgress.h"
+#include "AppInstallerStrings.h"
+#include "Microsoft/PredefinedInstalledSourceFactory.h"
+
+using namespace AppInstaller;
+using namespace AppInstaller::Repository;
+
+namespace winrt::Microsoft::Management::Deployment
+{
+    namespace ProgressSinkFactory
+    {
+        std::shared_ptr<AppInstaller::IProgressSink> CreatePackageCatalogProgressSink(std::string sourceType, std::function<void(double)> progressReporter, bool removeOperation)
+        {
+            if (sourceType.empty()
+                || Utility::CaseInsensitiveEquals( Repository::Microsoft::PredefinedInstalledSourceFactory::Type(), sourceType))
+            {
+                std::unordered_map<AppInstaller::ProgressType, double> progressWeights;
+
+                // There is no download operation for remove operation, so remove the download progress.
+                if (removeOperation)
+                {
+                    // it is percentage based progress.
+                    progressWeights.insert_or_assign(AppInstaller::ProgressType::Bytes, 0);
+                }
+                else
+                {
+                    // Add/Update operation has two progress types:
+                    // 1. Bytes for downloading index and
+                    // 2. Percent for index installation.
+                    progressWeights.insert_or_assign(AppInstaller::ProgressType::Bytes, 0.7);
+                    progressWeights.insert_or_assign(AppInstaller::ProgressType::Percent, 0.3);
+                }
+
+                return std::make_shared<PreIndexedPackageCatalogProgressSink>(progressWeights, progressReporter);
+            }
+            else
+            {
+                return std::make_shared<CompletionOnlyProgressSink>(progressReporter);
+            }
+        }
+    }
+
+    CompletionOnlyProgressSink::CompletionOnlyProgressSink(std::function<void(double)> progressReporter) :
+        m_progressReporter(progressReporter)
+    {
+    }
+
+    void CompletionOnlyProgressSink::OnProgress(uint64_t current, uint64_t maximum, AppInstaller::ProgressType type)
+    {
+        UNREFERENCED_PARAMETER(current);

```

Comment:
We prefer to leave the parameters unnamed rather than use `UNREFERENCED_PARAMETER`.